### PR TITLE
BB-353 - support s3:ObjectRestore:Post operation types in lifecycle

### PR DESCRIPTION
**Pull request by welansari**
Issue: [BB-353](https://scality.atlassian.net/browse/BB-353)

### CODE REVIEW

**Review by ghost — APPROVED** [review]
LGTM, aligned with what I found, thanks for adding a test as well!

File: extensions/lifecycle/LifecycleQueuePopulator.js [hunk 1]
```diff
@@ -225,7 +225,7 @@ class LifecycleQueuePopulator extends QueuePopulatorExtension {
         const value = JSON.parse(entry.value);
 
         const operation = value.originOp;
-        if (operation !== 's3:ObjectRestore') {
+        if (!['s3:ObjectRestore', 's3:ObjectRestore:Post'].includes(operation)) {
```
Comment (francoisferrand) [inline extensions/lifecycle/LifecycleQueuePopulator.js:231]: a comment explaining that this s3:ObjectRestore value is used for backwards compatibility with older cloudserver would be nice

**Review by francoisferrand — APPROVED** [review]
Should target 8.5, since cloudserver 8.6 with the breaking change is used in Zenko 2.5 along with backbeat 8.5

**Pull request merged**